app/src/test/java/com/example/smartsight/LocationSharingTest.kt

```diff
@@ -0,0 +1,303 @@
+package com.example.smartsight
+
+import android.content.ContentResolver
+import android.content.Context
+import android.content.SharedPreferences
+import android.database.Cursor
+import android.location.Location
+import android.os.Build
+import android.provider.ContactsContract
+import android.telephony.SmsManager
+import android.widget.Toast
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.Priority
+import com.google.android.gms.tasks.OnFailureListener
+import com.google.android.gms.tasks.OnSuccessListener
+import com.google.android.gms.tasks.Task
+import com.google.gson.Gson
+import io.mockk.*
+import org.junit.*
+import org.robolectric.annotation.Config
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertTrue
+
+/**
+ * Unit tests for the LocationSharing utility functions.
+ *
+ * These tests focus on non-UI logic such as:
+ * - Parsing phone numbers
+ * - Reading and saving contact lists
+ * - Sending SMS messages
+ * - Fetching location and handling success/failure gracefully
+ *
+ * All Android components are mocked using MockK to allow
+ * JVM-only testing (no device or emulator needed).
+ */
+@Config(sdk = [Build.VERSION_CODES.S])
+class LocationSharingTest {
+
+    private lateinit var context: Context
+    private lateinit var mockPrefs: SharedPreferences
+    private lateinit var mockEditor: SharedPreferences.Editor
+
+    @Before
+    fun setup() {
+        // Prepare mock context and shared preferences before each test
+        context = mockk(relaxed = true)
+        mockPrefs = mockk(relaxed = true)
+        mockEditor = mockk(relaxed = true)
+
+        // Mock SharedPreferences read/write chain
+        every { context.getSharedPreferences(any(), any()) } returns mockPrefs
+        every { mockPrefs.edit() } returns mockEditor
+        every { mockEditor.putString(any(), any()) } returns mockEditor
+        every { mockEditor.apply() } just runs
+    }
+
+    /**
+     * parseNumber() should correctly extract numeric digits from a formatted string.
+     */
+    @Test
+    fun `parseNumber extracts digits correctly`() {
+        val input = "John (+91 98765 43210)"
```

Comment:
I would not recommend using a 10 digit Indian Phone Number. It may belong to someone and I dont like having it here. replace it with a other 11 digit code if the function allows it